src/main/java/com/surrealdb/refactor/driver/WsPlaintextConnection.java

```diff
@@ -95,9 +108,12 @@ public QueryBlockResult query(String query, List<Param> params) {
                                         processedOuterResults[i] = val;
                                     }
                                 } else {
+                                    // exception is now thrown if results are not a Json Element
+                                    // instead of Json Array
+
                                     throw new SurrealDBUnimplementedException(
```

Comment:
This part of the code can be turned into a protocol exception. It will be triggered if the response is an error, so technically it is unimplemented. But the ticket it is pointing to is being solve by you 🙌

Reply:
okay sounds good. it's up to you on which you would prefer.

Reply:
I would like it to be a protocol exception, please :)

Reply:
done